### PR TITLE
Changelog for 0.6.0-beta

### PR DESCRIPTION
**Pull request by mkam**

### CODE REVIEW

**Comment by mkam** [comment]
Thanks @lornasong, your additions look good to me! I'm going to manually squash all these commits because we have merge and squash disabled. I'm pretty sure I disabled that too because of some shenanigans that caused when merging to enterprise.

**Review by wilkermichael — APPROVED** [review]

**Review by wilkermichael — APPROVED** [review]

**Comment by lornasong** [comment]
Rebased to latest on main. Added (and squashed) another entry in the deprecation section:

> Deprecate starting up CTS with no command in favor of starting up CTS using the new start command [https://github.com/hashicorp/consul-terraform-sync/issues/866]

Language edits welcome!

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,37 @@
...
+* Fix issue where the main function for running the CTS binary was only using a logger with default settings [[GH-789](https://github.com/hashicorp/consul-terraform-sync/issues/789)]
+* Fix issue where `syslog.enabled` was not automatically set to true when other `syslog` configs were configured [[GH-815](https://github.com/hashicorp/consul-terraform-sync/pull/815)]
+* Fix issue where deleting task run events used a read-lock instead of write-lock [[GH-787](https://github.com/hashicorp/consul-terraform-sync/pull/787)]
+
+DEPRECATIONS:
+* **(Enterprise Only)** Deprecate `terraform_version` task configuration in favor of `terraform_cloud_workspace.terraform_version` task configuration [[GH-790](https://github.com/hashicorp/consul-terraform-sync/issues/790)]
+* **(Enterprise Only)** Deprecate `license_path` configuration in favor of `license.path` configuration [[GH-810](https://github.com/hashicorp/consul-terraform-sync/issues/810)]
+* Deprecate starting up CTS with no command in favor of starting up CTS using the new `start` command [[GH-866](https://github.com/hashicorp/consul-terraform-sync/issues/866)]
```
Comment (wilkermichael) [inline CHANGELOG.md:33]:
```suggestion
* Deprecate starting CTS as a daemon using no CLI command in favor of starting CTS as a daemon using the new `start` command [[GH-866](https://github.com/hashicorp/consul-terraform-sync/issues/866)]
```

Comment (wilkermichael) [reply CHANGELOG.md:33]: @lornasong what do you think of this?

Comment (lornasong) [reply CHANGELOG.md:33]: Looks good to me, thank you. Will commit suggestion, locally squash, and then force push

Comment (wilkermichael) [reply CHANGELOG.md]: sounds good!

**Pull request merged**